lib/Doctrine/Common/Proxy/AbstractProxyFactory.php

```diff
@@ -93,12 +100,19 @@
      * @param \Doctrine\Common\Proxy\ProxyGenerator                     $proxyGenerator
      * @param \Doctrine\Common\Persistence\Mapping\ClassMetadataFactory $metadataFactory
      * @param bool|int                                                  $autoGenerate
+     *
+     * @throws \Doctrine\Common\Proxy\Exception\InvalidArgumentException When auto generate mode is not valid.
      */
     public function __construct(ProxyGenerator $proxyGenerator, ClassMetadataFactory $metadataFactory, $autoGenerate)
     {
         $this->proxyGenerator  = $proxyGenerator;
         $this->metadataFactory = $metadataFactory;
-        $this->autoGenerate    = (bool)$autoGenerate;
+
+        if ( ! in_array($autoGenerate, self::AUTOGENERATE_MODES, false)) {
```

Comment:
You don't need false here as it's default behavior.

Reply:
It is the default value, but I wanted to be explicit that we allow other values

Reply:
And why not casting it to int and making a strict comparison?

Reply:
Normally I don't set properties until they are validated. But I guess in this case makes no difference as it's the constructor